x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list.tsx

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CriteriaWithPagination,
+  EuiAvatar,
+  EuiBadge,
+  EuiBasicTable,
+  EuiButtonIcon,
+  EuiDescriptionList,
+  EuiEmptyPrompt,
+  EuiFacetButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiScreenReaderOnly,
+  EuiText,
+  EuiToolTip,
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui';
+
+import type { HorizontalAlignment } from '@elastic/eui';
+import React, { memo, useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import styled from 'styled-components';
+import { FormattedDate } from '../../../../common/components/formatted_date';
+import {
+  ActionDetails,
+  LogsEndpointActionResponse,
+  EndpointActivityLogAction,
+} from '../../../../../common/endpoint/types';
+import type { EndpointActionListRequestQuery } from '../../../../../common/endpoint/schema/actions';
+import { AdministrationListPage } from '../../../components/administration_list_page';
+import { ManagementEmptyStateWrapper } from '../../../components/management_empty_state_wrapper';
+import { useGetEndpointActionList } from '../../../hooks';
+import { OUTPUT_MESSAGES, TABLE_COLUMN_NAMES, UX_MESSAGES } from '../translations';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { useTestIdGenerator } from '../../../hooks/use_test_id_generator';
+
+// Truncated usernames
+const StyledFacetButton = styled(EuiFacetButton)`
+  .euiText {
+    margin-top: 0.38rem;
+    overflow-y: visible !important;
+  }
+`;
+
+export const ResponseActionsList = memo<
+  EndpointActionListRequestQuery & { hideHeader?: boolean; hideHostNameColumn?: boolean }
```

Comment:
seems like you are defining props to this component that are not being used... can you `Pick<>` out of `EndpointActionListRequestQuery` only the ones you are allowing to be be used in the component?

Reply:
Yeah, I'll update this once I've the date filter working. 👍